lib/Cmts.ml

```diff
@@ -59,15 +59,14 @@ let is_adjacent src (l1 : Location.t) (l2 : Location.t) =
     semicolon. If it is the case, comments attached to the following item
     should be kept after the infix symbol. *)
 let infix_symbol_before src (loc : Location.t) =
-  let pos_cnum = loc.loc_end.pos_cnum - 1 in
-  let loc_end = {loc.loc_end with pos_cnum} in
-  match Source.position_before src loc_end with
-  | Some loc_start ->
-      if loc_start.pos_cnum < loc.loc_end.pos_cnum then
-        let str = Source.string_at src loc_start loc.loc_end in
-        String.equal str ";" || Ast.String_id.is_infix str
-      else false
-  | None -> false
+  match Source.last_token_before src loc.loc_start with
+  | Some
+      ( ( SEMI | INFIXOP0 _ | INFIXOP1 _ | INFIXOP2 _ | INFIXOP3 _
```

Comment:
Can you explain this list of token ?

Reply:
I added a comment saying it corresponds to `Ast.String_id.is_infix`